components/gitpod-protocol/src/util/censor.ts

```diff
@@ -0,0 +1,11 @@
+/**
+ * Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+export function censor<T>(obj: T, k: keyof T): T {
```

Comment:
IMO this should _not_ be part of protocol, but is a service (`server`) concern.

Reply:
What is the correct place to share code between DB and server, if not protocol?

I was imitating other utils like `deep-partial.ts`, but I'm also happy to duplicate the `censor` function and have it just in server and in DB.

Reply:
Removed `censor` from protocol again (instead the code is now copy/pasted and specialized in ProjectDB).